utils/testgen.py

```diff
@@ -237,6 +237,11 @@ def pytest_generate_tests(metafunc):
             # Skip unwanted types
             continue
 
+        if hasattr(metafunc.function, 'meta'):
```

Comment:
It also occurs to me that if the meta marker code made it simple to be able to get a metadict, as though the meta mark has already been processed, that would really help this out. This this could start out as `test_flags = metamaker(metafunc.function)['test_flags'] or []` and we wouldn't need to getattr so much. I think that's out of the scope of this PR, but might be nice anyway. :)